resteasy-client-vertx/src/main/java/org/jboss/resteasy/client/jaxrs/engines/vertx/VertxClientHttpEngine.java

```diff
@@ -0,0 +1,283 @@
+        this.vertx = null;
+        this.httpClient = client;
+    }
+
+    @Override
+    public <T> Future<T> submit(final ClientInvocation request,
+                                final boolean buffered,
+                                final InvocationCallback<T> callback,
+                                final ResultExtractor<T> extractor) {
+        CompletableFuture<T> future = submit(request).thenCompose(response -> {
+            CompletableFuture<T> tmp = new CompletableFuture<>();
+            vertx.executeBlocking(promise -> {
+                try {
+                    T result = extractor.extractResult(response);
+                    tmp.complete(result);
+                } catch (Exception e) {
+                    tmp.completeExceptionally(e);
+                }
+            }, ar -> {
+                //
+            });
+            return tmp;
+        });
+        if (callback != null) {
+            future = future.whenComplete((response, throwable) -> {
+                if (throwable != null) {
+                    callback.failed(throwable);
+                } else {
+                    callback.completed(response);
+                }
+            });
+        }
+        return future;
+    }
+
+    @Override
+    public <T> CompletableFuture<T> submit(final ClientInvocation request,
+                                           final boolean buffered,
+                                           final ResultExtractor<T> extractor,
+                                           final ExecutorService executorService) {
+        return submit(request).thenCompose(response -> {
+            CompletableFuture<T> tmp = new CompletableFuture<>();
+            executorService.execute(() -> {
+                try {
+                    T result = extractor.extractResult(response);
+                    tmp.complete(result);
+                } catch (Exception e) {
+                    tmp.completeExceptionally(e);
+                }
+            });
+            return tmp;
+        });
+    }
+
+    public CompletableFuture<ClientResponse> submit(final ClientInvocation request) {
+
+        HttpMethod method;
+        try {
+            method = HttpMethod.valueOf(request.getMethod());
+        } catch (IllegalArgumentException e) {
+            method = HttpMethod.OTHER;
+        }
+
+        Object entity = request.getEntity();
+        Buffer body;
+        if (entity != null) {
+            body = Buffer.buffer(requestContent(request));
+        } else {
+            body = null;
+        }
+
+        RequestOptions options = new RequestOptions();
+        MultiMap headers = MultiMap.caseInsensitiveMultiMap();
+        request.getHeaders().asMap().forEach(headers::add);
+        options.setHeaders(headers);
+        if (body != null) {
+            headers.set(HttpHeaders.CONTENT_LENGTH, "" + body.length());
+        }
+        options.addHeader(HttpHeaders.USER_AGENT.toString(), "Vertx");
```

Comment:
Should it be `Resteasy with Eclipse Vert.x` here?

Reply:
yes maybe, I must admit I set it to make a test pass :-)

Reply:
I haven't changed this because i think it does not make much a difference